LLVM doesn't build  on the first try when cross compilling to windows-gnu host

Running `./x.py build --host x86_64-pc-windows-gnu --target x86_64-pc-windows-gnu` from Linux will fail on building LLVM for Windows (so it's 2nd LLVM build) because somehow LLVM_ON_UNIX is defined:
```
/home/mateusz/Projects/rust/rust/src/llvm-project/llvm/include/llvm/Support/FileSystem.h:164:3: error: ‘uid_t’ does not name a type; did you mean ‘pid_t’?
  164 |   uid_t fs_st_uid = 0;
      |   ^~~~~
      |   pid_t
```

Magic happens when running the same command second time without cleaning build dir or changing anything. It will finish LLVM build and install the files.

Rust build will fail on `librustc_codegen_llvm` because of missing `shell32` but this could be unrelated (although [CMake should link it](https://github.com/llvm/llvm-project/blob/6d6babf7457c8128072cb23f079123fa76cafc5b/llvm/lib/Support/CMakeLists.txt#L8)).

--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -226,7 +226,7 @@ fn make_win_dist(
         let idx = line.find(':').unwrap();
         let key = &line[..idx];
         let trim_chars: &[_] = &[' ', '='];
-        let value = line[(idx + 1)..].trim_start_matches(trim_chars).split(';').map(PathBuf::from);
+        let value = env::split_paths(line[(idx + 1)..].trim_start_matches(trim_chars));
 
         if key == "programs" {
             bin_path.extend(value);

--- a/src/librustc_llvm/build.rs
+++ b/src/librustc_llvm/build.rs
@@ -151,6 +151,12 @@ fn main() {
             continue;
         }
 
+        // Include path contains host directory, replace it with target
+        if is_crossed && flag.starts_with("-I") {
+            cfg.flag(&flag.replace(&host, &target));
+            continue;
+        }
+
         cfg.flag(flag);
     }
 
@@ -189,6 +195,9 @@ fn main() {
 
     if !is_crossed {
         cmd.arg("--system-libs");
+    } else if target.contains("windows-gnu") {
+        println!("cargo:rustc-link-lib=shell32");
+        println!("cargo:rustc-link-lib=uuid");
     }
     cmd.args(&components);
 

